pkg/manifests/manifests.go

```diff
@@ -1604,23 +1629,18 @@ func (f *Factory) PrometheusK8s(grpcTLS *v1.Secret, trustedCABundleCM *v1.Config
 		return nil, err
 	}
 
-	telemetryEnabled := f.config.ClusterMonitoringConfiguration.TelemeterClientConfig.IsEnabled()
 	clusterID := f.config.ClusterMonitoringConfiguration.TelemeterClientConfig.ClusterID
-	if telemetryEnabled && f.config.RemoteWrite {
-
+	if telemetrySecret != nil {
 		selectorRelabelConfig, err := promqlgen.LabelSelectorsToRelabelConfig(f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.TelemetryMatches)
```

Comment:
Discussed separately: Future readers of this code would benefit from checking the actual config condition here (`if telemetryEnabled && f.config.RemoteWrite`) and assuming the secret was created instead of coupling this to the secret creating.